app/models/collaborator.rb

```diff
@@ -2,6 +2,6 @@ class Collaborator < ApplicationRecord
   belongs_to :user
   belongs_to :project
 
-  validates :user_id, presence: true#, uniqueness: true
-  validates :project_id, presence: true#, uniqueness: true
+  validates :user_id, presence: true, uniqueness: { scope: :project_id }
```

Comment:
would you be able to explain a little more about how this is working? I understand the end result, but could use clarification...
- why does the scoped uniqueness need to be on both user_id and project_id? would having it on just one still prevent duplicates?

Reply:
Yeah it just makes sure that a user_id project_id pair is unique. I could just have it on one. Just covering all the bases

Reply:
cool, thanks for the explanation